src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/JsonHubProtocol.cs

```diff
@@ -117,11 +120,9 @@ private HubMessage ParseMessage(TextReader textReader, IInvocationBinder binder)
                         switch (reader.TokenType)
                         {
                             case JsonToken.PropertyName:
-                                string memberName = reader.Value.ToString();
-
-                                switch (memberName)
                                 {
-                                    case TypePropertyName:
+                                    if (object.ReferenceEquals(reader.Value, TypePropertyName) || string.Equals(reader.Value.ToString(), TypePropertyName))
```

Comment:
Gross. I wish there was a way to force `switch` to use reference equality.

Reply:
You don't need `object.` here though, right? Can you remove that at least? Maybe wrap this whole thing up in a helper `FastEquals`?

Reply:
Yea let me do that.